index/src/main.rs

```diff
@@ -39,14 +42,19 @@ fn main() {
     }
 }
 
+// TODO ask about try_main
```

Comment:
do you want me to make a pr updating `main` and removing `try_main`?

Reply:
I'm not sure what update to `main` you have in mind. Why remove `try_main`?

Reply:
I meant having `main` return the `Result` but after thinking more about, it try_main seems clearer.

Reply:
`main` should never return Result because that would make the error get Debug-printed, whereas Display is what I want here.